lib/test/cdo/test_share_filtering.rb

```diff
@@ -298,4 +298,65 @@ def test_should_filter_program_playlab_only_with_indicator
     assert_equal true,
       ShareFiltering.should_filter_program(with_indicator, 'playlab')
   end
+
+  def test_extract_text_blockly_handles_deeply_nested_json
```

Comment:
I'm not sure these tests are necessary. They are basically just testing whether the max depth parameter works. Do they take a long time since they have to generate a long string?

Reply:
I ran these locally and they didn't take long at all. I don't have a strong preference!